### ISSUE TITLE
Make miners always able to build Mining Towers

### ISSUE THREAD

**Issue by WilliamWeberBerrutti** [OpenHV/OpenHV#1075]
![Screenshot_20240725_194244](https://github.com/user-attachments/assets/aa6d7a2d-04f5-4180-b3fa-ed8260c8bc8a)

It should be interesting if miners could always build Mining Towers.

For the top case, make units move away from it.

For the bottom one, make Miners collect all scrap below the Mining Tower spot and then deploy it.

I also noticed vehicles do not spawn scrap at Mining Towers spot. Is that intentional?

**Comment by OcelotWalrus** [comment]
> For the top case, make units move away from it.

That already happens.

I guess what could be made is just reduce the size of the footprint: instead of 2x3, it'd be something like 2x2, and the two upper cells would be passable "from behind" by units.

**Comment by Mailaender** [comment]
> I also noticed vehicles do not spawn scrap at Mining Towers spot. Is that intentional?

Yes, to avoid blocking the spot. Otherwise the miner would collet the scrap but not deploy.

**Comment by OcelotWalrus** [comment]
So do you agree with making the tower footprint 2x2 Mailaender?

**Comment by Mailaender** [comment]
No, but making the units move away is realistic. Collecting the gold ball is harder to do.

**Comment by OcelotWalrus** [comment]
You could technically just make the miner move to every cell of the future footprint, so it'll collect any possible gold balls/scraps if it can't be deployed.
